MAUI/Scheduler/reminders.md

```diff
@@ -0,0 +1,244 @@
+            <syncfusion:SfScheduler.AppointmentMapping>
+                <syncfusion:AppointmentMapping
+                    Subject="EventName"
+                    StartTime="From"
+                    EndTime="To"
+                    AppointmentBackground="Color"
+                    IsAllDay="IsAllDay"
+                    StartTimeZone="StartTimeZone"
+                    EndTimeZone="EndTimeZone"
+                    RecurrenceExceptionDates="RecurrenceExceptions"
+                    RecurrenceRule="RecurrenceRule"
+                    RecurrenceId="RecurrenceId"
+                    Reminders="Reminders">
+                    <syncfusion:AppointmentMapping.ReminderMapping>
+                        <syncfusion:ReminderMapping IsDismissed="Dismissed"
+                                                    TimeBeforeStart="TimeBeforeStart"/>
+                    </syncfusion:AppointmentMapping.ReminderMapping>
+                </syncfusion:AppointmentMapping>
+            </syncfusion:SfScheduler.AppointmentMapping>
+        </syncfusion:SfScheduler>
+{% endhighlight %}
+{% highlight c# tabtitle="ReminderViewModel.cs" %}
+public class ReminderViewModel 
+{
+  ...
+  public ObservableCollection<Event> Events { get; set; } = new ObservableCollection<Event>();
+  this.Events.Add(new Event()
+  {
+    From = DateTime.Now,
+    To = DateTime.Now.AddHours(1),
+    Color = new SolidColorBrush((Color)ColorConverter.ConvertFromString("#FF339933")),
+    EventName = "Conference",
+    Reminders = new ObservableCollection<Reminder>
+    {
+        new Reminder { TimeBeforeStart = new TimeSpan(0)},
+    }
+    });
+}
+{% endhighlight %}
+{% endtabs %}
+
+## ReminderAlertOpening event
+Scheduler notify the appointment's reminder by [ReminderAlertOpening] event before the appointment's start time. The [ReminderAlertOpeningEventArgs] has the following property,
+* [Reminders]: Gets a list of reminders that are used to notify the appointment reminders.
+
+{% tabs %}
+{% highlight c# tabtitle="MainWindow.xaml.cs" hl_lines="1" %}
+scheduler.ReminderAlertOpening += Scheduler_ReminderAlertOpening;
+
+private void Scheduler_ReminderAlertOpening(object sender, ReminderAlertOpeningEventArgs e)
+{
+    var reminders = e.Reminders;
+    e.Reminders[0].IsDismissed = true;
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Handling reminders for future appointment
+
+* If the Reminder alert time is less than the current time, then appointment reminder time will be calculated from the current time.
+
+* Future normal and recurring appointment reminder alert will be displayed as Reminder deliver time.
+
+## Handling reminders for overdue appointment
+
+* If a Non-recurring appointments reminder is in past, reminder alert as overdue 
+
+* If a recurring appointment is in the past such past occurrences do not have reminders. 
+
+* Span appointment < 24 hours - Shows alert time in overdue from appointment 
+
+* Overdue after recurrence end 
+
```

Comment:
## Handling reminders for future appointment

•	Future normal and recurring appointment reminder alert will be displayed as Reminder alert time. 
•	Reminder alert time will be calculated form appointment start time substrate with reminder time interval. For an example if appointment time is 3.30pm and reminder time interval is new Timespan (0,15,0) then reminder alter time is 3.15 pm. 
•	If the Reminder alert time is less than the current time, then appointment reminder time will be calculated from the current time. 


## Handling reminders for overdue appointment

•	If a Non-recurring appointments reminder is in past, reminder alert as overdue until dismissed.
•	Recurring appointment overdue after recurrence end until dismissed and past occurrences do not have reminders.
•	Recurrence appointment due calculated from current date occurrence and its in due until next occurrence reminder time. 
•	If the last occurrence in recurrence is series is get dismissed, then the pattern recurring appointment dismissed. 

## Handling reminder for Changed occurrence appointment

•	Once the changed occurrence is moved to the future, its reminder is re-created from the pattern.
•	When the changed occurrence is restored to a time in the past, the reminder is deleted. 
•	Past Changed occurrence reminders will be dismissed.

Handle Dismissed for reminder
•	Normal appointment directly dismissed using 
•	For recurrence appointment, if current occurrence need to dismiss then need to add changed occurrence for reminder occurrence dismissed, then current occurrence dismissed details get updated in underlying appointment or data source. 
•	If only occurrence dismissed, then the changed icon will not be updated for dismissed changed occurrence 
Handle Snooze for reminder

Reply:
Changes updated.